gk/main.c

```diff
@@ -2271,6 +2267,24 @@ gk_proc(void *arg)
 			 * check if it's expired.
 			 */
 			rte_prefetch_non_temporal(fe);
+
+			iter_count = 0;
+		} else
+			iter_count++;
+
+		process_pkts_front(port_front, port_back,
+			rx_queue_front, tx_queue_back,
+			lcore, &front_num_pkts, front_icmp_bufs,
+			instance, gk_conf);
+
+		process_pkts_back(port_back, port_front,
+			rx_queue_back, tx_queue_front, lcore,
+			&back_num_pkts, back_icmp_bufs, instance, gk_conf);
+
+		if (fe != NULL) {
+			flow_hash_val = rss_ip_flow_hf(&fe->flow, 0, 0);
+			rte_hash_prefetch_buckets(instance->ip_flow_hash_table,
+				flow_hash_val);
```

Comment:
We should only calculate the hash and issue these prefetches if `fe` is in use and expired. In other words, we no longer need `gk_flow_tbl_entry_scan()`; bring the code of `gk_flow_tbl_entry_scan()` into `gk_proc()`.